azurelinuxagent/ga/telemetry_service.py

```diff
@@ -81,18 +85,27 @@ def join(self):
     def stopped(self):
         return not self.should_run
 
-    def enqueue_event_func(self, event):
+    def enqueue_event(self, event):
         # Add event to queue and set event
-        self._queue.put(event)
+        if self.stopped():
+            raise ServiceStoppedError("{0} is stopped, not accepting anymore events".format(self.get_thread_name()))
+
+        # Queue.put() can block if the queue is full which can be an uninterruptible wait. Blocking for a max of
```

Comment:
Good to know!

Reply:
I will probably answer my own question here, but doesn't hurt to double check.

Why are we using no blocking for `get`, but blocking timeout for `put`? I assume it's because if `put` can't execute due to the queue being full, we expect it to be cleared out pretty frequently as we are sending out events. Is this true?

Reply:
For get, we're already using an `thread.Event` for synchronization, plus there are additional checks (like empty checks) in the code which take care of ensuring we dont try to fetch from the queue if its empty.

For put, as you mentioned, that is the expectation. And if the queue is still full, we just mark the service as stopped to ensure no new events are added either. 

Hope this clarifies your doubts

Reply:
All clear now, thanks!